Чистка кода

WolfManR opened the pull request.

WolfManR committed:
HRInPocket Program class code clear

WolfManR committed:
Переместил интеграцию сервисов в DI контейнер в класс DIConfiguration

WolfManR committed:
Переместил маппинг профиля пользователя из ProfileExtension в профиль Автомапера

WolfManR committed:
Remodel DTO Services

Infarh commented on src/HRInPocket/Services/HRInPocket.Interfaces/Services/Repository/IRepository.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace HRInPocket.Interfaces.Services.Repository
+{
+    public interface IRepository<TDto> where TDto: BaseDTO
```
Если в параметре типа используется ограничение на то, что TDto должно быть BaseDTO, то интерфейс следует обозвать IDtoRepository

Infarh commented on src/HRInPocket/Services/HRInPocket.Interfaces/Services/Repository/IResumeService.cs:
```diff
@@ -0,0 +1,32 @@
+
+        // Методы поиска резюме
+
+        Task SearchResumesAsync();
```
Фильтр в метод надо добавить...

WolfManR replied:
исправляется в пулле #81

Infarh commented on src/HRInPocket/Services/HRInPocket.Interfaces/Services/Repository/ITargetTaskService.cs:
```diff
@@ -0,0 +1,32 @@
+    /// <summary>
+    /// Сервис упраавления заданиями
+    /// </summary>
+    public interface ITargetTaskService : IRepository<TargetTaskDTO>
```
Нужны ещё методы запуска и остановки сервиса, а также свойство, характеризующее его текущее состояние. Либо, быть может, имеет смысл разделить на сервис, который чисто манипулирует сущностями Заданий (даёт посмотреть, добавить, удалить, найти по заданному критерию - по сути, репозиторий) и сервис, который выполняет эти задания (TaskRunner).

WolfManR replied:
исправляется в пулле #81

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/CompanyService.cs:
```diff
@@ -27,16 +29,18 @@ public CompanyService(IDataRepository<Company> dataProvider, IMapper mapper)
-                /*Логика фильтрации после понимания структуры фильтров*/ 
+            if (filter != null)
+            {
+                /*Логика фильтрации после понимания структуры фильтров*/
```
А не стоит ли эту логику прямо в этот класс фильтра и засунуть? Пусть фильтр занимается тем, что фильтрует переданный ему запрос и возвращает уже преобразованный - с наложенными на него фильтрами.

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/CompanyService.cs:
```diff
@@ -45,9 +49,9 @@ public async Task<PageCompanyDTO> GetCompanies(CompanyFilter filter = null)
+            return new PageDTOs<CompanyDTO>
             {
-                Companies = query.Select(q => _Mapper.Map<CompanyDTO>(q)),
+                Entities = query.Select(q => _Mapper.Map<CompanyDTO>(q)),
```
```suggestion
                Entities = query.Select(_Mapper.Map<CompanyDTO>),
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/DataRepository.cs:
```diff
@@ -177,14 +176,51 @@ public async Task<bool> RemoveAsync(Guid id)
             var table = _DbContext.Set<TEntity>();
 
             var item = await table.FirstOrDefaultAsync(s => s.Id == id);
-            if (item is null) 
+            if (item is null)
                 throw new ArgumentNullException(nameof(item));
 
             table.Remove(item);
```
Можно удалить сущность не загружая её из БД в контекст примерно следующим образом
```C#
_DbContext.Entity(new TEntity { Id = id }).State = EntityState.Deleted;
_DbContext.SaveChanges();
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/DataRepository.cs:
```diff
@@ -177,14 +176,51 @@ public async Task<bool> RemoveAsync(Guid id)
+        /// Создать диапазон объектов в базе данных
+        /// </summary>
+        /// <param name="items">Диапазон объектов</param>
+        public void CreateRange(ICollection<TEntity> items)
```
```suggestion
        public void CreateRange(IEnumerable<TEntity> items)
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/DataRepository.cs:
```diff
@@ -177,14 +176,51 @@ public async Task<bool> RemoveAsync(Guid id)
+        /// Создать диапазон объектов в базе данных
+        /// </summary>
+        /// <param name="items">Диапазон объектов</param>
+        public async Task CreateRangeAsync(ICollection<TEntity> items)
```
```suggestion
        public async Task CreateRangeAsync(IEnumerable<TEntity> items)
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/ResumeService.cs:
```diff
@@ -45,49 +52,53 @@ public async Task<PageResumeDTO> GetResumesAsync(ResumeFilter filter)
+            return new PageDTOs<ResumeDTO>
             {
-                Companies = query.Select(q => _Mapper.Map<ResumeDTO>(q)),
+                Entities = query.Select(q => _Mapper.Map<ResumeDTO>(q)),
```
```suggestion
                Entities = query.Select(_Mapper.Map<ResumeDTO>),
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/ResumeService.cs:
```diff
@@ -27,10 +31,13 @@ public ResumeService(IDataRepository<Resume> dataProvider, IMapper mapper)
-        public async Task<PageResumeDTO> GetResumesAsync(ResumeFilter filter)
+        public async Task<PageDTOs<ResumeDTO>> GetAllAsync(Filter filter)
         {
             var query = _DataProvider.GetQueryable();
```
Быть может это тоже стоит убрать в фильтр? Сделать универсальный базовый класс фильтра, способный накладывать постраничное разбиение на запрос.
Что-нибудь типа 
```C#
public async Task<(IQueryable<T> Query, int TotalCount)> Paging(IQueryable<T> query, int Page, int PageSize)
{
    var total_count = await query.CountAsync();
    var result = query.Skip(Page * PageSize).Take(PageSize);
    return (result, total_count);
}
```

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/ResumeService.cs:
```diff
@@ -45,49 +52,53 @@ public async Task<PageResumeDTO> GetResumesAsync(ResumeFilter filter)
+        public async Task<IEnumerable<ResumeDTO>> GetUserResumesAsync(Guid id) =>
+            (await _DataProvider.GetQueryableAsync())
+            .Where(r => r.Applicant.ProfileId == id.ToString())
+            .Select(r => _Mapper.Map<ResumeDTO>(r));
```
```suggestion
            .Select(_Mapper.Map<ResumeDTO>);
```

WolfManR committed:
Merge branch 'dev' into Tarxos/CodeClear

WolfManR committed:
code clear

WolfManR committed:
fix DataRepository

WolfManR committed:
fix naming of IRepository to IDtoRepository

WolfManR committed:
code cleanup

WolfManR committed:
refactoring of DtoRepository

WolfManR committed:
Replace repositories and its bases

WolfManR committed:
Linq Extensions refactoring

WolfManR committed:
filter update

Infarh committed:
Выгрузка заданий в профиле пользователя

Infarh committed:
Первое WebAPI - управление задачами - просмотр задач текущего пользвоателя

Infarh committed:
Определение перечня действий контроллера WebAPI управления задачами

Infarh committed:
Ещё заготовки контроллеров API

WolfManR committed:
Merge branch 'dev' into Tarxos/CodeClear

WolfManR committed:
comment IIS configuration

WolfManR committed:
remove fake logger error message on db initialization

WolfManR committed:
Assignments api

WolfManR committed:
experimental swagger optional route parameter fix

todo: work not correct, rework

WolfManR committed:
remove TestApiController

WolfManR committed:
update Assignments Api Controller

WolfManR committed:
Account Api

WolfManR committed:
remove obsollete code

WolfManR committed:
replace base injection of Automapper

Infarh commented on src/HRInPocket/Common/HRInPocket.Domain/Filters/Filter.cs:
```diff
@@ -0,0 +1,29 @@
+
+        public IQueryable<T> Paging<T>(IQueryable<T> query)
+        {
+            var (page, pageSize) = (Pages.PageNumber, Pages.PageSize);
```
Вот это вообще непонятно зачем. Зачем вводить локальные переменные?

WolfManR replied:
предложение решарпера, всегда просит разобрать модель на запчасти

WolfManR replied:
убрать? а как в решарпере убрать эту предложение?

Infarh commented on src/HRInPocket/Services/HRInPocket.DAL/Data/TestDbInitializer.cs:
```diff
@@ -26,13 +26,13 @@ public void Initialize()
-                    .InitTable(TestData.TargetTasks)
+                .InitTable(TestData.Addresses)
+                .InitTable(TestData.Specialties)
+                .InitTable(TestData.ActivityCategories)
+                .InitTable(TestData.TargetTasks)
 
-                    .InitTable(TestData.Tarifs)
-                    .InitTable(TestData.Price)
+                .InitTable(TestData.Tarifs)
+                .InitTable(TestData.Price)
```
Мне не нравится что внутри этих методов нельзя пользоваться логгером. Этот метод более нигде не применим. Может быть его реализовать просто как метод класса инициализатора и там будет тогда возможность добавить логирование?

WolfManR replied:
исправлено в коммитах f06ffff & 38ffda3

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/Base/DataRepository.cs:
```diff
@@ -0,0 +1,205 @@
+            _DbContext.Attach(item);
+            _DbContext.Update(item);
+            _DbContext.SaveChanges();
+            return true;
```
А этот метод вообще может вернуть false?

WolfManR replied:
перенаправить в #81

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/Base/DataRepository.cs:
```diff
@@ -0,0 +1,205 @@
+
+            var item = table.FirstOrDefault(s => s.Id == id);
+            if (item is null)
+                throw new ArgumentNullException(nameof(item));
```
А Вот тут стоило бы вернуть false

WolfManR replied:
перенаправить в #81

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/Base/DataRepository.cs:
```diff
@@ -0,0 +1,205 @@
+            var table = _DbContext.Set<TEntity>();
+            table.RemoveRange(items);
+            _DbContext.SaveChanges();
+            return true;
```
Тут тоже bool-результат бессмысленен.

WolfManR replied:
перенаправить в #81

Infarh commented on src/HRInPocket/Services/HRInPocket.Services/Repositories/ResumeService.cs:
```diff
@@ -0,0 +1,46 @@
+        public async Task<IEnumerable<ResumeDTO>> GetUserResumesAsync(Guid id) =>
+            (await _DataProvider.GetQueryableAsync())
+            .Where(r => r.Applicant.ProfileId == id.ToString())
+            .AsEnumerable()
+            .Select(_Mapper.Map<ResumeDTO>);
```
У AutoMapper'а есть пакет, позволяющий выполнять те же самые действия, но внутри запроса EF.
То ли AutoMapper.Extensions.ExpressionMapping, то ли AutoMapper.Data, то ли AutoMapper.Collection.EntityFrameworkCore
AutoMapper.AspNetCore.OData.EFCore

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Controllers/API/AccountApiController.cs:
```diff
@@ -0,0 +1,221 @@
+        public RegistrationException(UserData user) : base($"User with {user.email} already exist") => User = user;
+    }
+
+    public class LoginException : AuthorizationException
```
👍

WolfManR replied:
предложите чем-то уже сейчас дополнить?

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Controllers/API/AssignmentsApiControllers.cs:
```diff
@@ -0,0 +1,307 @@
+
+    #region Api Models
+
+    [ModelBinder(typeof(AssignmentTypeModelBinder))]
```
👍

WolfManR commented:
планы:
 - добавить логирование
 - вытащить модели и сервисы
 - скорректировать работу написанных сервисов
 - добавить логической функциональности сервисам и моделям

WolfManR committed:
Rework db initialize process

WolfManR committed:
replace IDbInitializer in to interfaces

WolfManR committed:
feedback blank

WolfManR committed:
Feedback prototype

WolfManR committed:
replace Assignments structs

WolfManR committed:
Refactor replacements of FeedBack, Account, Assignment api models

WolfManR committed:
cut notification implementation from AccountApi

WolfManR committed:
LogEvents id's

WolfManR committed:
add creation time generation to FeedbackRequest record

WolfManR committed:
Account Api logger injection

Add LogEvents id's for failure in Account Api,
rework Auth Service

WolfManR committed:
rework Assignment system

WolfManR committed:
ApplicantsApi Blank

WolfManR committed:
fix Creating test service NotifyService

WolfManR committed:
Update Assignments api

Logging implemented,
Add Assign to applicant behavior

WolfManR committed:
Update check IsApplicantExist

WolfManR committed:
add Feedback Assignment

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Infrastructure/Models/LogEvents.cs:
```diff
@@ -0,0 +1,31 @@
+﻿namespace HRInPocket.Infrastructure.Models
+{
+    public static class LogEvents
+    {
+        public const int GenerateItems = 1000;
+        public const int ListItems     = 1001;
+        public const int GetItem       = 1002;
+        public const int InsertItem    = 1003;
+        public const int UpdateItem    = 1004;
+        public const int DeleteItem    = 1005;
+
+        public const int TestItem      = 3000;
+
+        public const int GenerateItemsFailure = 4000;
+        public const int ListItemsNotFound = 4001;
+        public const int GetItemNotFound    = 4002;
+        public const int InsertItemFailure    = 4003;
+        public const int UpdateItemNotFound = 4004;
+        public const int DeleteItemFailure = 4005;
+
+        public const int AccountRegistration = 5000;
+        public const int AccountLogin = 5001;
+        public const int AccountLogout = 5002;
+        
+        public const int AccountRegistrationFailure = 5010;
+        public const int AccountLoginFailure = 5011;
+        public const int AccountLogoutFailure = 5012;
+
+        public const int MailSending = 9000;
+    }
```
Может быть стоит переделать этот класс в enum?

WolfManR replied:
оказалось нельзя

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Infrastructure/Models/Records/Assignments/Assignment.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace HRInPocket.Infrastructure.Models.Records.Assignments
+{
+    public abstract record Assignment(string place_name)
+    {
+        public long id { get; set; }
+        public Guid applicant_id { get; private set; }
+        public DateTime date_added { get; } = DateTime.Now;
+        public int number_of_responses { get; set; }
+        public int number_of_invitations { get; set; }
```
Record со свойствами с возможностью записи?

WolfManR replied:
временные модели, по причине характера использования изменить нельзя

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Infrastructure/Models/Records/FeedbackRequest.cs:
```diff
@@ -0,0 +1,12 @@
+{
+    public record FeedbackRequest(string name, string email, string phone_number, string message)
+    {
+        public long id { get; set; }
+        public Guid AssignedApplicant { get; set; }
+        public DateTime RequestCreation { get; } = DateTime.Now;
+        public DateTime ApplicantAssignedDate { get; set; }
```
Аналогично...

WolfManR replied:
временные модели, по причине характера использования изменить нельзя

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Infrastructure/Models/Records/NotifyUser.cs:
```diff
@@ -0,0 +1,9 @@
+{
+    public record NotifyUser(string email)
+    {
+        public bool EmailNotify { get; set; }
+        public bool SmsNotify { get; set; }
+        public bool PushNotify { get; set; }
```
```suggestion
        public bool EmailNotify { get; init; }
        public bool SmsNotify { get; init; }
        public bool PushNotify { get; init; }
```

Infarh commented on src/HRInPocket/UI/WEB/HRInPocket/Controllers/API/AssignmentsApiControllers.cs:
```diff
@@ -0,0 +1,248 @@
+        public AssignmentsApiControllers(ILogger<AssignmentsApiControllers> logger)
+        {
+            _logger = logger;
+            var authService = new AuthService();
+            _applicantsManager = new ApplicantManagerService(authService);
+            _assignmentsManager = new AssignmentsManagerService(_applicantsManager);
```
Стоит разместить в контейнере Ioc

WolfManR replied:
я помню

WolfManR committed:
replace Services to Services, add its to DI Container

WolfManR committed:
replace LogEvents

WolfManR committed:
replace Authorize Exceptions to Domain

WolfManR committed:
replace AssignmentType enum

WolfManR committed:
replace refactoring Json Returns Types

WolfManR committed:
replace refactor test records

WolfManR committed:
refactor structs

WolfManR committed:
Merge branch 'dev' into Tarxos/CodeClear

WolfManR committed:
refactor merge

remove not needed classes,
append api route company to WebApi,
refactor Shoping Service,
update services registration,
replace api controllers code to its place,
refactor Web UI Project configuration
